iron-form.html

```diff
@@ -0,0 +1,146 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-ajax/iron-ajax.html">
+
+<!--
+An `iron-form` is a form element that can validate and submit any custom
+elements that have an 'iron-form-behavior`, as well as any native HTML elements.
+-->
+
+<script>
+
+  Polymer({
+
+    is: 'iron-form',
+
+    extends: 'form',
+
+    /**
+     * Fired after the form is submitted.
+     *
+     * @event iron-form-submit
+     */
+
+    /**
+    * Fired after the form is submitted and a response is received.
+    *
+    * @event iron-form-response
+    */
+
+    /**
+     * Fired after the form is submitted and an error is received.
+     *
+     * @event iron-form-error
+     */
+    properties: {
+
+      /**
+       * Contains all the registered form elements and their values that
+       * will be used to submit the form.
+       */
+      jsonData: {
+        type: Object,
+        notify: true
+      }
+
+    },
+
+    listeners: {
+      'iron-form-element-register': '_registerElement',
+      'submit': 'submit'
+    },
+
+    ready: function() {
+      // Object that handles the ajax form submission request.
+      this._requestBot = document.createElement('iron-ajax');
+      this._requestBot.addEventListener('response', this._handleFormResponse.bind(this));
+      this._requestBot.addEventListener('error', this._handleFormError.bind(this));
+
+      // Holds all the custom elements registered with this form.
+      this._customElements = [];
+    },
+
+    /**
+     * Called to submit the form.
+     */
+    submit: function(event) {
+      var valid = true;
+
+      for (var el, i = 0; el = this._customElements[i], i < this._customElements.length; i++) {
+        valid = el.validate() && valid;
+      }
+
+      if (!valid) {
+        return false;
+      }
+
+      this.serialize();
+
+      this._requestBot.url = this.action;
+      this._requestBot.method = this.method;
+      this._requestBot.params = this.jsonData;
+
+      if (this.method == 'POST') {
+        this._requestBot.body = JSON.stringify(this.jsonData);
+      }
+
+      this._requestBot.generateRequest();
+      this.fire('iron-form-submit');
+
+      // Don't perform a page refresh.
+      if (event)
```

Comment:
Minor style detail, but this conditional should probably have opening and closing `{ ... }`